### ISSUE TITLE
Area2d Black Icon on Mac OSX 10.13

### ISSUE THREAD

**Issue by stonekase** [godotengine/godot#21615]
**Godot version:**

3.1 custom build

**OS/device including version:**

Mac Book  pro. Nvidia GT 750M

**Issue description:**

Area2d has black Icons in a scene
**Steps to reproduce:**

**Minimal reproduction project:**

add area2d as a child of a position2d with both as children of a kinematic2d
<img width="373" alt="screen shot 2018-08-30 at 10 19 49 am" src="https://user-images.githubusercontent.com/5037778/44886942-68b29600-acc1-11e8-84fe-689c9d492e98.png">

<img width="665" alt="screen shot 2018-08-30 at 10 23 47 am" src="https://user-images.githubusercontent.com/5037778/44886954-7ec05680-acc1-11e8-92cf-f52fa6b8d14c.png">

**Comment by akien-mga** [comment]
Can you share the scene which triggers this issue? It looks like it tries to load `res://Sprites/Fists/right_fist.png` as icon for the `right_fist` Area2D, and fails.

Probably related to the new custom node icons, CC @willnationsdev.

**Comment by stonekase** [comment]
@akien-mga  below is the scene


[playerone.tscn.zip](https://github.com/godotengine/godot/files/2341805/playerone.tscn.zip)

**Comment by akien-mga** [comment]
There's indeed some
```
__meta__ = {
"_editor_icon": SubResource( 5 )
}
```
on those two nodes.

Did you set an editor icon manually, and later on delete it? Or did it happen by itself?

**Comment by stonekase** [comment]
Yes I actually did that, then re-imported the new Icon.

**Comment by willnationsdev** [comment]
@akien-mga sorry, I'll have some time to help take a look at this today I think.

**Comment by willnationsdev** [comment]
@stonekase, try as I might, I can't seem to replicate the effect here. I discovered a separate bug where commenting out the icon path from a GDScript isn't removing the icon from the scene's data (and therefore the scene dock), and I can also make the icon vanish completely by moving the icon's location outside of Godot and not fixing the scene dependencies when I re-enter. But I haven't been able to make the icon become all-black. I'm using Linux Mint, but I'm not sure if that's relevant.

**Comment by willnationsdev** [comment]
I've identified why this happens. If you create a script with a class name, but no icon, it will set an `_editor_icon` value of null. When you then clear the script, this null value isn't removed, and saving the scene results in an all black inlined image resource getting used for the null value. Upon reloading the scene (I restarted the editor), I started getting the black icon. I have a fix in mind and will send in a PR later today when I have more time.

**Comment by willnationsdev** [comment]
Well, nevermind. What I thought was causing it is not actually the cause. The "fixes" in the linked PR don't actually solve this problem like I thought they would. When debugging with these changes, even though the `SceneTreeDock::_script_created` method concluded with me assigning a valid base class's icon to an icon-less script class's node, saving the scene was still setting the _editor_icon value to null and generating an all-black image / image texture.

There has to be something else during the scene saving process that is setting it to null and generating the inlined scene content.

**Comment by willnationsdev** [comment]
When I looked at the callstack for the `p_image.is_null()` error, it showed that the error was related to the dynamic font of the ScriptCreateDialog's title bar not generating an image properly, so that seems related to a separate bug entirely.

![screenshot from 2018-09-01 22-37-54](https://user-images.githubusercontent.com/16217563/44951946-b6510f00-ae37-11e8-86cd-84839570e95c.png)

**Comment by willnationsdev** [comment]
I have found an entirely separate callback sequence that triggers the same `p_image.is_null()` error, and it is again related to the `get_char_size(...)` method in DynamicFont (seems to be the common denominator between call stacks):

![screenshot from 2018-09-02 11-00-33](https://user-images.githubusercontent.com/16217563/44958042-d2889680-ae9f-11e8-8d99-1a529128a99a.png)

**Comment by willnationsdev** [comment]
@akien-mga I'm not sure if this bug is actually closed by #21717. My last comment on here indicated that the error was originating from the images in a font somewhere.

**Comment by akien-mga** [comment]
@willnationsdev Are you sure this `p_image.is_null()` error is related to this bug though? The callstacks don't seem related at all.

**Comment by willnationsdev** [comment]
@akien-mga That's what I'm saying. You closed the Issue because of the editor icon refactor, but I'm saying that I don't think the editor icon refactor would have fixed it, since the callstack seems to indicate otherwise.

**Comment by willnationsdev** [comment]
@akien-mga Oh, sorry, I realized that I said in my PR that it fixes the bug. It fixes a DIFFERENT bug, that is also brought up during this Issue.

**Comment by willnationsdev** [comment]
Oh wait...nevermind. I'm an idiot. The original bug is for the black box thing, not the p_image.is_null() error. We need to just make a separate bug Issue for that. Ignore me.

**Comment by akien-mga** [comment]
Yeah that's what I meant, `p_image.is_null()` is a normal error if you're trying to load an image which doesn't exist, as was the case here due to a bug in editor icons. If the editor icons bug is resolved, it's fine and the error message should no longer be printed.

If you still get this error locally, it's likely a different bug.
